src/stim/simulators/tableau_simulator.cc

```diff
@@ -734,7 +742,7 @@ void TableauSimulator::ensure_large_enough_for_qubits(size_t num_qubits) {
     inv_state.expand(num_qubits);
 }
 
-void TableauSimulator::sample_stream(FILE *in, FILE *out, SampleFormat format, bool interactive, std::mt19937_64 &rng) {
+void TableauSimulator::sample_stream(FILE *in, FILE *out, SampleFormat format, bool interactive, const std::mt19937_64 &rng) {
```

Comment:
Is this supposed to have been changed to const? The semantics of the method are changed pretty drastically by this change. This may have broken some of the fuzz tests so that they only check one case.

This method is clearly a sortof leftover from the old way the code was using RNGs by passing them around. It probably needs to be refactored much more extremely in order to make it work with the new way where the simulators own their RNGs. Either the RNG state needs to be written back into this value as the method returns, or the parameter needs to be refactored to be like a seed or something.

Reply:
Will do, see other comment.